src/main/kotlin/com/hibob/academy/filter/AuthenticationFilter.kt

```diff
@@ -21,18 +20,23 @@ class AuthenticationFilter: ContainerRequestFilter {
     override fun filter(requestContext: ContainerRequestContext) {
 
         if (requestContext.uriInfo.path != CREATE_TOKEN_PATH) {
-            verify(requestContext.cookies[COOKIE_NAME]?.value)
+            verify(requestContext.cookies[COOKIE_NAME]?.value, requestContext)
             logger.info("Authorized successfully")
         }
     }
 
-    fun verify(cookie: String?): Jws<Claims> {
+    fun verify(cookie: String?, requestContext:ContainerRequestContext) {
         cookie?.let {
-            return try {
+            try {
                 Jwts.parser().setSigningKey(SECRET_KEY).parseClaimsJws(it)
             } catch (e: Exception) {
-                throw Exception("unauthorized error: ${e.message}")
+                requestContext.abortWith(Response.status(Response.Status.UNAUTHORIZED).build())
+                logger.warn("Authorized failure - exception: $e")
```

Comment:
best would be to extract these lines as a function. notice that you logged a different level message in the two instances that you abort the request - here a warning and on line 39 an info. if you extract to a function you ensure the same behavior in both instances